### ISSUE TITLE
AAX parameters not initialized correctly

### ISSUE THREAD

**Issue by pressplay-music** [iPlug2/iPlug2#1111]
If you `Set` a parameter value in your plug-in constructor, this value will be ignored in AAX versions of the plug-in. This does not happen with VST or audio units.

Reproduction:

```
MyPlugin::MyPlugin(...)
{
    ...
    GetParam(0)->InitDouble(...);
    GetParam(0)->Set(ValueOtherThanDefault);
}
```

`ValueOtherThanDefault` will be ignored. This scenario may sound a bit exotic. In my case, I am setting a user-defined initial value that is loaded from a json file stored on disk. This value is not the parameter default value (if you double-click on the corresponding control you will reset to the default value, not the user-defined init value).

I believe I know the reason for this behaviour and how to fix it. But I wanted to discuss this first before submitting a pull request.

So in `IPlugAAX.cpp` there is a function `EffectInit()` where `AAX_CParameter`s are created. I think we need to also call `pAAXParam->SetValueWithDouble(pParam->Value());` there.

Thoughts?

**Comment by TBProAudio** [comment]
Right, other initial values ​​as default do not work with the current AAX implementation.

I use the following fix in IPlugAAX.cpp:

```
    switch (pParam->Type())
    {
      case IParam::kTypeDouble:
      {
        pAAXParam = new AAX_CParameter<double>(pParamIDStr->Get(),
                                          AAX_CString(pParam->GetName()),
#if 0
                                          pParam->GetDefault(),
#else
                                          pParam->Value(),                      // TBPROAUDIO 15072024 Support startup/default preset
#endif
                                          AAX_CIPlugTaperDelegate<double>(*pParam),
                                          AAX_CUnitDisplayDelegateDecorator<double>(AAX_CNumberDisplayDelegate<double>(), AAX_CString(pParam->GetLabel())),
                                          pParam->GetCanAutomate());
        
        pAAXParam->SetNumberOfSteps(128); // TODO: check this https://developer.digidesign.com/index.php?L1=5&L2=13&L3=56
        pAAXParam->SetType(AAX_eParameterType_Continuous);

        break;
      }
```

Same for int and enaum/bool. 
I hope this helps.
Thomas

**Comment by pressplay-music** [comment]
This does not look right to me. You lose your IParam default value by doing this. If your goal is to replace the default value I would rather call `GetParam(yourParamIndex)->SetDefault(yourNewDefault)` in your plug-in constructor.

I still believe `pAAXParam->SetValueWithDouble(pParam->Value());` is the right approach.
